In [43]:
import pandas as pd

In [44]:
df = pd.read_csv('data/wine_cleared.csv')

In [45]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129971 entries, 0 to 129970
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Unnamed: 0             129971 non-null  int64  
 1   country                129908 non-null  object 
 2   description            129971 non-null  object 
 3   designation            129971 non-null  object 
 4   points                 129971 non-null  int64  
 5   price                  129971 non-null  float64
 6   province               129908 non-null  object 
 7   region_1               129971 non-null  object 
 8   taster_name            129971 non-null  object 
 9   taster_twitter_handle  129971 non-null  object 
 10  title                  129971 non-null  object 
 11  variety                129970 non-null  object 
 12  winery                 129971 non-null  object 
dtypes: float64(1), int64(2), object(10)
memory usage: 12.9+ MB


In [46]:
df.drop(columns=['Unnamed: 0'], inplace=True)

In [47]:
df.head()

country                                        description  \
0     Italy  Aromas include tropical fruit, broom, brimston...   
1  Portugal  This is ripe and fruity, a wine that is smooth...   
2        US  Tart and snappy, the flavors of lime flesh and...   
3        US  Pineapple rind, lemon pith and orange blossom ...   
4        US  Much like the regular bottling from 2012, this...   

                          designation  points      price           province  \
0                        Vulkà Bianco      87  35.363389  Sicily & Sardinia   
1                            Avidagos      87  15.000000              Douro   
2                             unknown      87  14.000000             Oregon   
3                Reserve Late Harvest      87  13.000000           Michigan   
4  Vintner's Reserve Wild Child Block      87  65.000000             Oregon   

              region_1         taster_name taster_twitter_handle  \
0                 Etna       Kerin O’Keefe          @kerinokeefe   
1              unknown          Roger Voss            @vossroger   
2    Willamette Valley        Paul Gregutt           @paulgwine    
3  Lake Michigan Shore  Alexander Peartree               unknown   
4    Willamette Valley        Paul Gregutt           @paulgwine    

                                               title         variety  \
0                  Nicosia 2013 Vulkà Bianco  (Etna)     White Blend   
1      Quinta dos Avidagos 2011 Avidagos Red (Douro)  Portuguese Red   
2      Rainstorm 2013 Pinot Gris (Willamette Valley)      Pinot Gris   
3  St. Julian 2013 Reserve Late Harvest Riesling ...        Riesling   
4  Sweet Cheeks 2012 Vintner's Reserve Wild Child...      Pinot Noir   

                winery  
0              Nicosia  
1  Quinta dos Avidagos  
2            Rainstorm  
3           St. Julian  
4         Sweet Cheeks

In [48]:
df['price_round'] = df['price'].round().astype('int')

In [49]:
df['year'] = df['title'].str.findall('\d{4}').str.get(0).astype('int', copy=False, errors='ignore')

In [50]:
df.head()

country                                        description  \
0     Italy  Aromas include tropical fruit, broom, brimston...   
1  Portugal  This is ripe and fruity, a wine that is smooth...   
2        US  Tart and snappy, the flavors of lime flesh and...   
3        US  Pineapple rind, lemon pith and orange blossom ...   
4        US  Much like the regular bottling from 2012, this...   

                          designation  points      price           province  \
0                        Vulkà Bianco      87  35.363389  Sicily & Sardinia   
1                            Avidagos      87  15.000000              Douro   
2                             unknown      87  14.000000             Oregon   
3                Reserve Late Harvest      87  13.000000           Michigan   
4  Vintner's Reserve Wild Child Block      87  65.000000             Oregon   

              region_1         taster_name taster_twitter_handle  \
0                 Etna       Kerin O’Keefe          @kerinokeefe   
1              unknown          Roger Voss            @vossroger   
2    Willamette Valley        Paul Gregutt           @paulgwine    
3  Lake Michigan Shore  Alexander Peartree               unknown   
4    Willamette Valley        Paul Gregutt           @paulgwine    

                                               title         variety  \
0                  Nicosia 2013 Vulkà Bianco  (Etna)     White Blend   
1      Quinta dos Avidagos 2011 Avidagos Red (Douro)  Portuguese Red   
2      Rainstorm 2013 Pinot Gris (Willamette Valley)      Pinot Gris   
3  St. Julian 2013 Reserve Late Harvest Riesling ...        Riesling   
4  Sweet Cheeks 2012 Vintner's Reserve Wild Child...      Pinot Noir   

                winery  price_round  year  
0              Nicosia           35  2013  
1  Quinta dos Avidagos           15  2011  
2            Rainstorm           14  2013  
3           St. Julian           13  2013  
4         Sweet Cheeks           65  2012

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129971 entries, 0 to 129970
Data columns (total 14 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   country                129908 non-null  object 
 1   description            129971 non-null  object 
 2   designation            129971 non-null  object 
 3   points                 129971 non-null  int64  
 4   price                  129971 non-null  float64
 5   province               129908 non-null  object 
 6   region_1               129971 non-null  object 
 7   taster_name            129971 non-null  object 
 8   taster_twitter_handle  129971 non-null  object 
 9   title                  129971 non-null  object 
 10  variety                129970 non-null  object 
 11  winery                 129971 non-null  object 
 12  price_round            129971 non-null  int32  
 13  year                   125362 non-null  object 
dtypes: float64(1), int32(1), int64(1), o

In [52]:
df['is_usa'] = df['country'].apply(lambda x: 1 if x == 'US' else 0)

###  Задание 2.1 ###

Выбрать из списка две самых популярных (помимо США) страны, производящих вино.

***Ответ - France, Italy***

In [53]:
df['country'].value_counts()

US                        54504
France                    22093
Italy                     19540
Spain                      6645
Portugal                   5691
Chile                      4472
Argentina                  3800
Austria                    3345
Australia                  2329
Germany                    2165
New Zealand                1419
South Africa               1401
Israel                      505
Greece                      466
Canada                      257
Hungary                     146
Bulgaria                    141
Romania                     120
Uruguay                     109
Turkey                       90
Slovenia                     87
Georgia                      86
England                      74
Croatia                      73
Mexico                       70
Moldova                      59
Brazil                       52
Lebanon                      35
Morocco                      28
Peru                         16
Ukraine                      14
Serbia  

###  Задание 2.2 ###

Создать бинарные признаки `is_france`, `is_italy` наподобие признака `is_usa`.

В ответ вписать результат выполнения кода `data['is_france'].sum()` и `data['is_italy'].sum()`

***Ответ - 22093, 19540***

In [54]:
df['is_france'] = df['country'].apply(lambda x: 1 if x == 'France' else 0)
df['is_italy'] = df['country'].apply(lambda x: 1 if x == 'Italy' else 0)
df['is_france'].sum()

22093

In [55]:
df['is_italy'].sum()

19540

###  Задание 2.3 ###

Создать новый бинарный признак `old_wine`, где значение **1** будет обозначать, что вино старше 2010 года.

В ответ вписать результат выполнения кода `data['old_wine'].sum()`

***Ответ - 39781***

In [60]:
df['old_wine'] = df['year'].apply(lambda x: 1 if pd.to_numeric(x) < 2010 else 0)

In [63]:
df['old_wine'].sum()

39781

###  Задание 2.7 (Самопроверка) ###

Создать новый признак `locality` из признака `title`, который будет обозначать название долины/местности производства вина.

Например, в названии вина **Rainstorm 2013 Pinot Gris (Willamette Valley)** `locality` будет **Willamette Valley**. В названии **Tandem 2011 Ars In Vitro Tempranillo-Merlot (Navarra)**&nbsp;&mdash; **Navarra**.

In [75]:
df['locality'] = df['title'].str.findall(r'\(.+\)').str.get(0).str.replace('\(|\)', '', regex=True)

In [76]:
df.head(20)

country                                        description  \
0       Italy  Aromas include tropical fruit, broom, brimston...   
1    Portugal  This is ripe and fruity, a wine that is smooth...   
2          US  Tart and snappy, the flavors of lime flesh and...   
3          US  Pineapple rind, lemon pith and orange blossom ...   
4          US  Much like the regular bottling from 2012, this...   
5       Spain  Blackberry and raspberry aromas show a typical...   
6       Italy  Here's a bright, informal red that opens with ...   
7      France  This dry and restrained wine offers spice in p...   
8     Germany  Savory dried thyme notes accent sunnier flavor...   
9      France  This has great depth of flavor with its fresh ...   
10         US  Soft, supple plum envelopes an oaky structure ...   
11     France  This is a dry wine, very spicy, with a tight, ...   
12         US  Slightly reduced, this wine offers a chalky, t...   
13      Italy  This is dominated by oak and oak-driven aromas...   
14         US  Building on 150 years and six generations of w...   
15    Germany  Zesty orange peels and apple notes abound in t...   
16  Argentina  Baked plum, molasses, balsamic vinegar and che...   
17  Argentina  Raw black-cherry aromas are direct and simple ...   
18      Spain  Desiccated blackberry, leather, charred wood a...   
19         US  Red fruit aromas pervade on the nose, with cig...   

                                          designation  points      price  \
0                                        Vulkà Bianco      87  35.363389   
1                                            Avidagos      87  15.000000   
2                                             unknown      87  14.000000   
3                                Reserve Late Harvest      87  13.000000   
4                  Vintner's Reserve Wild Child Block      87  65.000000   
5                                        Ars In Vitro      87  15.000000   
6                                             Belsito      87  16.000000   
7                                             unknown      87  24.000000   
8                                               Shine      87  12.000000   
9                                         Les Natures      87  27.000000   
10                                     Mountain Cuvée      87  19.000000   
11                                            unknown      87  30.000000   
12                                            unknown      87  34.000000   
13                                              Rosso      87  35.363389   
14                                            unknown      87  12.000000   
15                                              Devon      87  24.000000   
16                                              Felix      87  30.000000   
17                                Winemaker Selection      87  13.000000   
18  Vendimia Seleccionada Finca Valdelayegua Singl...      87  28.000000   
19                                            unknown      87  32.000000   

             province             region_1         taster_name  \
0   Sicily & Sardinia                 Etna       Kerin O’Keefe   
1               Douro              unknown          Roger Voss   
2              Oregon    Willamette Valley        Paul Gregutt   
3            Michigan  Lake Michigan Shore  Alexander Peartree   
4              Oregon    Willamette Valley        Paul Gregutt   
5      Northern Spain              Navarra   Michael Schachner   
6   Sicily & Sardinia             Vittoria       Kerin O’Keefe   
7              Alsace               Alsace          Roger Voss   
8         Rheinhessen              unknown  Anna Lee C. Iijima   
9              Alsace               Alsace          Roger Voss   
10         California          Napa Valley      Virginie Boone   
11             Alsace               Alsace          Roger Voss   
12         California     Alexander Valley      Virginie Boone   
13  Sicily & Sardinia                 Etna       Kerin O